sql/core/src/test/scala/org/apache/spark/sql/StatisticsCollectionSuite.scala

```diff
@@ -650,4 +651,17 @@ class StatisticsCollectionSuite extends StatisticsCollectionTestBase with Shared
       }
     }
   }
+
+  test("SPARK-30903: Fail fast on duplicate columns when analyze columns") {
+    val table = "test_table"
+    withTable(table) {
+      sql(s"""
+           |CREATE TABLE $table (value string, name string)
+           |USING PARQUET""".stripMargin)
+      val errorMsg = intercept[ParseException] {
+        sql(s"ANALYZE TABLE $table COMPUTE STATISTICS FOR COLUMNS value, name, name, value")
```

Comment:
I think we need to take care of case-sensitivity and need some tests for that. We can use some methods to check the name duplication in `SchemaUtils`?

Reply:
Make sense about case-sensitivity.

And it seems like we may could reuse `checkColumnNameDuplication` in `SchemaUtils` but needs slightly change since we'd expect `ParseException` here rather than `AnalysisException`.

Reply:
Yea, we need some refactoring for this use... the refactoring itself looks ok to me.